docs/odcs.rst

```diff
@@ -24,84 +24,188 @@ idea is to not require the user to provide a yum repo url when building
 container images.  Instead, OSBS will use ODCS to generate a yum repo with the
 relevant signed packages based on user configuration.
 
+Concepts
+--------
+
+Signing Intent
+""""""""""""""
+
+When an ODCS compose is used, the idea of a signing intent is used. This refers
+to the signature of the content to be included in the container image. For
+instance, an environment may provide the following intents: ``release``,
+``beta``, and ``unsigned``. Each one of those intents is then mapped to a list
+of signing keys. These signing keys are then used during compose creation. The
+packages to be included must have been signed by either one of the keys listed.
+In the example above, the intents could be mapped to the following keys::
+
+    # Only include packages that have been signed by "my-release-key"
+    release -> my-release-key
+    # Include packages that have been signed by either "my-beta-key" or
+    # "my-release-key"
+    beta -> my-beta-key, my-release-key
+    # Do not check signature of packages - may include unsigned packages
+    unsigned -> <empty>
+
+*This mapping will be defined in reactor-config-secret which will allow updates
+to be reflected instantly for automatically triggered builds from an existing
+BuildConfig.*
+
+The definition of intents also describes the restrictive order, which will be
+enforced when building layered images. For instance, consider the case of two
+images, X and Y. Y uses X as its parent image (FROM X). If image X was built
+with "beta" intent, image Y's intent can only be "beta" or "unsigned". If the
+repo for image Y has it configured to use "release" intent, this value will be
```

Comment:
Meaning the dist-git repository?

Reply:
Correct. Changed it to "dist-git repo" for clarification.